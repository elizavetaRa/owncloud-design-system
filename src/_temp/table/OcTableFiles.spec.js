import { mount } from "@vue/test-utils"
import { DateTime } from "luxon"

import Table from "./OcTableFiles.vue"

const getCurrentDate = () => {
  return DateTime.fromJSDate(new Date()).minus({ days: 1 }).toFormat("EEE, dd MMM yyyy HH:mm:ss")
}

const fields = ["name", "size", "mdate", "sdate", "ddate", "actions", "owner", "sharedWith"]

const sharedWith = [
  {
    id: "bob",
    username: "bob",
    displayName: "Bob",
    avatar:
      "https://images.unsplash.com/photo-1610216705422-caa3fcb6d158?ixid=MXwxMjA3fDB8MHxzZWFyY2h8MTB8fGZhY2V8ZW58MHwyfDB8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=60",
  },
  {
    id: "marie",
    username: "marie",
    displayName: "Marie",
    avatar:
      "https://images.unsplash.com/photo-1584308972272-9e4e7685e80f?ixid=MXwxMjA3fDB8MHxzZWFyY2h8Mzh8fGZhY2V8ZW58MHwyfDB8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=60",
  },
  {
    id: "john",
    username: "john",
    displayName: "John Richards Emperor of long names",
  },
]

const owner = [
  {
    id: "bob",
    username: "bob",
    displayName: "Bob",
    avatar:
      "https://images.unsplash.com/photo-1610216705422-caa3fcb6d158?ixid=MXwxMjA3fDB8MHxzZWFyY2h8MTB8fGZhY2V8ZW58MHwyfDB8&ixlib=rb-1.2.1&auto=format&fit=crop&w=800&q=60",
  },
]

const indicators = [
  {
    id: "files-sharing",
    label: "Shared with other people",
    visible: true,
    icon: "group",
    handler: (resource, indicatorId) =>
      alert(`Resource: ${resource.name}, indicator: ${indicatorId}`),
  },
  {
    id: "file-link",
    label: "Shared via link",
    visible: true,
    icon: "link",
  },
]

const resourcesWithAllFields = [
  {
    id: "forest",
    name: "forest.jpg",
    path: "images/nature/forest.jpg",
    preview: "https://cdn.pixabay.com/photo/2015/09/09/16/05/forest-931706_960_720.jpg",
    indicators,
    type: "file",
    size: "111000234",
    mdate: getCurrentDate(),
    sdate: getCurrentDate(),
    ddate: getCurrentDate(),
    owner,
    sharedWith,
  },
  {
    id: "notes",
    name: "notes.txt",
    path: "/Documents/notes.txt",
    icon: "text",
    indicators,
    type: "file",
    size: "big",
    mdate: getCurrentDate(),
    sdate: getCurrentDate(),
    ddate: getCurrentDate(),
    sharedWith,
    owner,
  },
  {
    id: "documents",
    name: "Documents",
    path: "/Documents",
    icon: "folder",
    indicators,
    type: "folder",
    size: "-1",
    mdate: getCurrentDate(),
    sdate: getCurrentDate(),
    ddate: getCurrentDate(),
    sharedWith,
    owner,
  },
]

describe("OcTableFiles", () => {
  // As we do not care about design here, we can render the full set of fields
  // Since we are using mount, use it only once to save some time
  const wrapper = mount(Table, {
    propsData: {
      resources: resourcesWithAllFields,
      selection: [],
      slots: {
        status: "<div class='status-slot'>Hello world!</div>",
      },
    },
  })

  it("displays all fields", () => {
    for (const field of fields) {
      expect(wrapper.findAll(".oc-table-header-cell-" + field).length).toEqual(1)
      expect(wrapper.findAll(".oc-table-data-cell-" + field).length).toEqual(
        resourcesWithAllFields.length
      )
    }

    expect(wrapper).toMatchSnapshot()
  })

  it("emits showDetails event when clicking on the button in actions column", () => {
    wrapper.find(".oc-table-data-cell-actions .oc-table-files-btn-show-details").vm.$emit("click")

    expect(wrapper.emitted().showDetails).toBeTruthy()
  })

  it("emits showDetails event when clicking on the row", async () => {
    wrapper.find(".oc-tbody-tr").trigger("click")

    await wrapper.vm.$nextTick()

    expect(wrapper.emitted().showDetails.length).toEqual(2)
  })

  it("formats the resource size", () => {
    expect(wrapper.find(".oc-tbody-tr-forest .oc-table-data-cell-size").text()).toEqual("105.9 MB")
    expect(wrapper.find(".oc-tbody-tr-documents .oc-table-data-cell-size").text()).toEqual("")
    expect(wrapper.find(".oc-tbody-tr-notes .oc-table-data-cell-size").text()).toEqual("?")
  })

  it("selects resources", async () => {
    wrapper.find(".oc-table-files-select-all .oc-checkbox").trigger("click")
    wrapper.find(".oc-tbody-tr-documents .oc-checkbox").trigger("click")
    expect(wrapper.emitted().select.length).toBe(2)
  })

  it("de-selects all resources", () => {
    const wrapperSelected = mount(Table, {
      propsData: {
        resources: resourcesWithAllFields,
        selection: ["forest", "notes", "documents"],
      },
    })

    wrapperSelected.find(".oc-table-files-select-all .oc-checkbox").trigger("click")
    expect(wrapperSelected.emitted().select[0][0].length).toBe(0)
  })

  it("emits fileClick", () => {
    wrapper.find(".oc-tbody-tr-forest .oc-resource-name").trigger("click")

    expect(wrapper.emitted().fileClick[0][0].name).toMatch("forest.jpg")
  })
})
