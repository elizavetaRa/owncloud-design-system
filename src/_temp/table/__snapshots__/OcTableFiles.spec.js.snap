// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcTableFiles displays all fields 1`] = `
<table class="oc-table oc-table-sticky" slots="[object Object]">
  <thead class="oc-thead">
    <tr>
      <th class="oc-th oc-table-cell oc-table-cell-align-left oc-table-cell-align-bottom oc-table-cell-width-shrink oc-text-nowrap oc-table-header-cell oc-table-header-cell-select" style="top: 0px;">
        <div class="oc-table-files-select-all"><label for="oc-table-files-select-all" class="oc-cursor-pointer"><input id="oc-table-files-select-all" type="checkbox" name="checkbox" aria-label="Select all resources" class="oc-checkbox oc-checkbox-m" value="">
            <!---->
          </label></div>
      </th>
      <th class="oc-th oc-table-cell oc-table-cell-align-left oc-table-cell-align-bottom oc-table-cell-width-expand oc-text-nowrap oc-table-header-cell oc-table-header-cell-name" style="top: 0px;">
        Name
      </th>
      <th class="oc-th oc-table-cell oc-table-cell-align-right oc-table-cell-align-bottom oc-table-cell-width-auto oc-text-nowrap oc-table-header-cell oc-table-header-cell-size" style="top: 0px;">
        Size
      </th>
      <th class="oc-th oc-table-cell oc-table-cell-align-right oc-table-cell-align-bottom oc-table-cell-width-auto oc-text-nowrap oc-table-header-cell oc-table-header-cell-sharedWith" style="top: 0px;">
        Shared with
      </th>
      <th class="oc-th oc-table-cell oc-table-cell-align-right oc-table-cell-align-bottom oc-table-cell-width-auto oc-text-nowrap oc-table-header-cell oc-table-header-cell-owner" style="top: 0px;">
        Owner
      </th>
      <th class="oc-th oc-table-cell oc-table-cell-align-right oc-table-cell-align-bottom oc-table-cell-width-auto oc-text-nowrap oc-table-header-cell oc-table-header-cell-mdate" style="top: 0px;">
        Modified
      </th>
      <th class="oc-th oc-table-cell oc-table-cell-align-right oc-table-cell-align-bottom oc-table-cell-width-auto oc-text-nowrap oc-table-header-cell oc-table-header-cell-sdate" style="top: 0px;">
        Shared
      </th>
      <th class="oc-th oc-table-cell oc-table-cell-align-right oc-table-cell-align-bottom oc-table-cell-width-auto oc-text-nowrap oc-table-header-cell oc-table-header-cell-ddate" style="top: 0px;">
        Deleted
      </th>
      <th class="oc-th oc-table-cell oc-table-cell-align-right oc-table-cell-align-bottom oc-table-cell-width-auto oc-text-nowrap oc-table-header-cell oc-table-header-cell-actions" style="top: 0px;">
        Actions
      </th>
    </tr>
  </thead>
  <tbody>
    <tr class="oc-tbody-tr oc-tbody-tr-forest" style="height: 64px;">
      <td class="oc-td oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-shrink oc-table-data-cell oc-table-data-cell-select"><label for="oc-table-files-select-forest" class="oc-cursor-pointer"><input id="oc-table-files-select-forest" type="checkbox" name="checkbox" aria-label="Select" class="oc-checkbox oc-checkbox-l" value="[object Object]">
          <!---->
        </label></td>
      <td class="oc-td oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-expand oc-text-truncate oc-table-data-cell oc-table-data-cell-name">
        <div class="oc-resource oc-text-overflow"><img src="https://cdn.pixabay.com/photo/2015/09/09/16/05/forest-931706_960_720.jpg" aria-hidden="true" width="40" height="40" class="oc-resource-preview">
          <div class="oc-resource-details oc-text-overflow"><button class="oc-text-overflow oc-button oc-button-m oc-button-justify-content-center oc-button-gap-undefined oc-button-raw oc-button-raw-color-default">
              <div resource-path="images/nature/forest.jpg" resource-name="forest.jpg" class="oc-resource-name oc-text-truncate">
                <!----><span class="oc-resource-basename">forest</span><span class="oc-resource-extension">.jpg</span>
              </div>
            </button>
            <div class="oc-resource-indicators">
              <div class="oc-status-indicators"><button aria-label="Shared with other people" class="oc-status-indicators-indicator oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-raw oc-button-raw-color-default" id="files-sharing" uk-tooltip="Shared with other people"><span aria-label="icon" class="oc-icon oc-icon-m oc-icon-passive" aria-hidden="true">function SvgGroup(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgGroup.name})
  );
}</span></button><span aria-label="Shared via link" class="oc-status-indicators-indicator oc-icon oc-icon-m oc-icon-passive" id="file-link" tabindex="-1" uk-tooltip="Shared via link">function SvgLink(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgLink.name})
  );
}</span></div>
            </div>
          </div>
        </div>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-size"><span class="oc-resource-size">105.9 MB</span></td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-sharedWith">
        <div uk-tooltip="Bob, Marie, John Richards Emperor of long names" class="oc-avatar-group oc-table-files-people oc-avatar-group-stacked">
          <div class="vue-avatar--wrapper oc-avatar" style="display: flex; width: 30px; height: 30px; border-radius: 50%; line-height: 31px; font-weight: bold; align-items: center; justify-content: center; text-align: center; user-select: none;" alt="Bob"><img src="https://images.unsplash.com/photo-1610216705422-caa3fcb6d158?ixid=MXwxMjA3fDB8MHxzZWFyY2h8MTB8fGZhY2V8ZW58MHwyfDB8&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=800&amp;q=60" style="display: none;"> <span style="display: none;"></span></div>
          <div class="vue-avatar--wrapper oc-avatar" style="display: flex; width: 30px; height: 30px; border-radius: 50%; line-height: 31px; font-weight: bold; align-items: center; justify-content: center; text-align: center; user-select: none;" alt="Marie"><img src="https://images.unsplash.com/photo-1584308972272-9e4e7685e80f?ixid=MXwxMjA3fDB8MHxzZWFyY2h8Mzh8fGZhY2V8ZW58MHwyfDB8&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=800&amp;q=60" style="display: none;"> <span style="display: none;"></span></div>
          <div class="vue-avatar--wrapper oc-avatar" style="display: flex; width: 30px; height: 30px; border-radius: 50%; align-items: center; justify-content: center; text-align: center; user-select: none; background-color: rgb(63, 81, 181); font: sans-serif; color: rgb(143, 161, 255);" alt="John Richards Emperor of long names">
            <!----> <span>J</span>
          </div>
          <!---->
          <!---->
        </div>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-owner">
        <div uk-tooltip="Bob" class="oc-avatar-group oc-table-files-people">
          <div class="vue-avatar--wrapper oc-avatar" style="display: flex; width: 30px; height: 30px; border-radius: 50%; line-height: 31px; font-weight: bold; align-items: center; justify-content: center; text-align: center; user-select: none;" alt="Bob"><img src="https://images.unsplash.com/photo-1610216705422-caa3fcb6d158?ixid=MXwxMjA3fDB8MHxzZWFyY2h8MTB8fGZhY2V8ZW58MHwyfDB8&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=800&amp;q=60" style="display: none;"> <span style="display: none;"></span></div>
          <!---->
          <!---->
        </div>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-mdate">
        1 day ago
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-sdate">
        1 day ago
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-ddate">
        1 day ago
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-actions">
        <div class="oc-table-files-actions"> <button aria-label="Show details" class="oc-table-files-btn-show-details oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-raw oc-button-raw-color-default"><span aria-label="icon" class="oc-icon oc-icon-m oc-icon-system" aria-hidden="true">function SvgMore_vert(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgMore_vert.name})
  );
}</span></button></div>
      </td>
    </tr>
    <tr class="oc-tbody-tr oc-tbody-tr-notes" style="height: 64px;">
      <td class="oc-td oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-shrink oc-table-data-cell oc-table-data-cell-select"><label for="oc-table-files-select-notes" class="oc-cursor-pointer"><input id="oc-table-files-select-notes" type="checkbox" name="checkbox" aria-label="Select" class="oc-checkbox oc-checkbox-l" value="[object Object]">
          <!---->
        </label></td>
      <td class="oc-td oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-expand oc-text-truncate oc-table-data-cell oc-table-data-cell-name">
        <div class="oc-resource oc-text-overflow"><span aria-label="icon" class="oc-icon oc-icon-l oc-icon-file-type" aria-hidden="true">function SvgText(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgText.name})
  );
}</span>
          <div class="oc-resource-details oc-text-overflow"><button class="oc-text-overflow oc-button oc-button-m oc-button-justify-content-center oc-button-gap-undefined oc-button-raw oc-button-raw-color-default">
              <div resource-path="/Documents/notes.txt" resource-name="notes.txt" class="oc-resource-name oc-text-truncate">
                <!----><span class="oc-resource-basename">notes</span><span class="oc-resource-extension">.txt</span>
              </div>
            </button>
            <div class="oc-resource-indicators">
              <div class="oc-status-indicators"><button aria-label="Shared with other people" class="oc-status-indicators-indicator oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-raw oc-button-raw-color-default" id="files-sharing" uk-tooltip="Shared with other people"><span aria-label="icon" class="oc-icon oc-icon-m oc-icon-passive" aria-hidden="true">function SvgGroup(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgGroup.name})
  );
}</span></button><span aria-label="Shared via link" class="oc-status-indicators-indicator oc-icon oc-icon-m oc-icon-passive" id="file-link" tabindex="-1" uk-tooltip="Shared via link">function SvgLink(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgLink.name})
  );
}</span></div>
            </div>
          </div>
        </div>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-size"><span class="oc-resource-size">?</span></td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-sharedWith">
        <div uk-tooltip="Bob, Marie, John Richards Emperor of long names" class="oc-avatar-group oc-table-files-people oc-avatar-group-stacked">
          <div class="vue-avatar--wrapper oc-avatar" style="display: flex; width: 30px; height: 30px; border-radius: 50%; line-height: 31px; font-weight: bold; align-items: center; justify-content: center; text-align: center; user-select: none;" alt="Bob"><img src="https://images.unsplash.com/photo-1610216705422-caa3fcb6d158?ixid=MXwxMjA3fDB8MHxzZWFyY2h8MTB8fGZhY2V8ZW58MHwyfDB8&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=800&amp;q=60" style="display: none;"> <span style="display: none;"></span></div>
          <div class="vue-avatar--wrapper oc-avatar" style="display: flex; width: 30px; height: 30px; border-radius: 50%; line-height: 31px; font-weight: bold; align-items: center; justify-content: center; text-align: center; user-select: none;" alt="Marie"><img src="https://images.unsplash.com/photo-1584308972272-9e4e7685e80f?ixid=MXwxMjA3fDB8MHxzZWFyY2h8Mzh8fGZhY2V8ZW58MHwyfDB8&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=800&amp;q=60" style="display: none;"> <span style="display: none;"></span></div>
          <div class="vue-avatar--wrapper oc-avatar" style="display: flex; width: 30px; height: 30px; border-radius: 50%; align-items: center; justify-content: center; text-align: center; user-select: none; background-color: rgb(63, 81, 181); font: sans-serif; color: rgb(143, 161, 255);" alt="John Richards Emperor of long names">
            <!----> <span>J</span>
          </div>
          <!---->
          <!---->
        </div>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-owner">
        <div uk-tooltip="Bob" class="oc-avatar-group oc-table-files-people">
          <div class="vue-avatar--wrapper oc-avatar" style="display: flex; width: 30px; height: 30px; border-radius: 50%; line-height: 31px; font-weight: bold; align-items: center; justify-content: center; text-align: center; user-select: none;" alt="Bob"><img src="https://images.unsplash.com/photo-1610216705422-caa3fcb6d158?ixid=MXwxMjA3fDB8MHxzZWFyY2h8MTB8fGZhY2V8ZW58MHwyfDB8&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=800&amp;q=60" style="display: none;"> <span style="display: none;"></span></div>
          <!---->
          <!---->
        </div>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-mdate">
        1 day ago
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-sdate">
        1 day ago
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-ddate">
        1 day ago
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-actions">
        <div class="oc-table-files-actions"> <button aria-label="Show details" class="oc-table-files-btn-show-details oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-raw oc-button-raw-color-default"><span aria-label="icon" class="oc-icon oc-icon-m oc-icon-system" aria-hidden="true">function SvgMore_vert(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgMore_vert.name})
  );
}</span></button></div>
      </td>
    </tr>
    <tr class="oc-tbody-tr oc-tbody-tr-documents" style="height: 64px;">
      <td class="oc-td oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-shrink oc-table-data-cell oc-table-data-cell-select"><label for="oc-table-files-select-documents" class="oc-cursor-pointer"><input id="oc-table-files-select-documents" type="checkbox" name="checkbox" aria-label="Select" class="oc-checkbox oc-checkbox-l" value="[object Object]">
          <!---->
        </label></td>
      <td class="oc-td oc-table-cell oc-table-cell-align-left oc-table-cell-align-middle oc-table-cell-width-expand oc-text-truncate oc-table-data-cell oc-table-data-cell-name">
        <div class="oc-resource oc-text-overflow"><span aria-label="icon" class="oc-icon oc-icon-l oc-icon-file-type" aria-hidden="true">function SvgFolder(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgFolder.name})
  );
}</span>
          <div class="oc-resource-details oc-text-overflow"><a class="oc-text-overflow">
              <div resource-path="/Documents" resource-name="Documents" class="oc-resource-name oc-text-truncate">
                <!----><span class="oc-resource-basename">Documents</span>
                <!---->
              </div>
            </a>
            <div class="oc-resource-indicators">
              <div class="oc-status-indicators"><button aria-label="Shared with other people" class="oc-status-indicators-indicator oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-raw oc-button-raw-color-default" id="files-sharing" uk-tooltip="Shared with other people"><span aria-label="icon" class="oc-icon oc-icon-m oc-icon-passive" aria-hidden="true">function SvgGroup(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgGroup.name})
  );
}</span></button><span aria-label="Shared via link" class="oc-status-indicators-indicator oc-icon oc-icon-m oc-icon-passive" id="file-link" tabindex="-1" uk-tooltip="Shared via link">function SvgLink(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgLink.name})
  );
}</span></div>
            </div>
          </div>
        </div>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-size"><span class="oc-resource-size"></span></td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-sharedWith">
        <div uk-tooltip="Bob, Marie, John Richards Emperor of long names" class="oc-avatar-group oc-table-files-people oc-avatar-group-stacked">
          <div class="vue-avatar--wrapper oc-avatar" style="display: flex; width: 30px; height: 30px; border-radius: 50%; line-height: 31px; font-weight: bold; align-items: center; justify-content: center; text-align: center; user-select: none;" alt="Bob"><img src="https://images.unsplash.com/photo-1610216705422-caa3fcb6d158?ixid=MXwxMjA3fDB8MHxzZWFyY2h8MTB8fGZhY2V8ZW58MHwyfDB8&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=800&amp;q=60" style="display: none;"> <span style="display: none;"></span></div>
          <div class="vue-avatar--wrapper oc-avatar" style="display: flex; width: 30px; height: 30px; border-radius: 50%; line-height: 31px; font-weight: bold; align-items: center; justify-content: center; text-align: center; user-select: none;" alt="Marie"><img src="https://images.unsplash.com/photo-1584308972272-9e4e7685e80f?ixid=MXwxMjA3fDB8MHxzZWFyY2h8Mzh8fGZhY2V8ZW58MHwyfDB8&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=800&amp;q=60" style="display: none;"> <span style="display: none;"></span></div>
          <div class="vue-avatar--wrapper oc-avatar" style="display: flex; width: 30px; height: 30px; border-radius: 50%; align-items: center; justify-content: center; text-align: center; user-select: none; background-color: rgb(63, 81, 181); font: sans-serif; color: rgb(143, 161, 255);" alt="John Richards Emperor of long names">
            <!----> <span>J</span>
          </div>
          <!---->
          <!---->
        </div>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-owner">
        <div uk-tooltip="Bob" class="oc-avatar-group oc-table-files-people">
          <div class="vue-avatar--wrapper oc-avatar" style="display: flex; width: 30px; height: 30px; border-radius: 50%; line-height: 31px; font-weight: bold; align-items: center; justify-content: center; text-align: center; user-select: none;" alt="Bob"><img src="https://images.unsplash.com/photo-1610216705422-caa3fcb6d158?ixid=MXwxMjA3fDB8MHxzZWFyY2h8MTB8fGZhY2V8ZW58MHwyfDB8&amp;ixlib=rb-1.2.1&amp;auto=format&amp;fit=crop&amp;w=800&amp;q=60" style="display: none;"> <span style="display: none;"></span></div>
          <!---->
          <!---->
        </div>
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-mdate">
        1 day ago
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-sdate">
        1 day ago
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-ddate">
        1 day ago
      </td>
      <td class="oc-td oc-table-cell oc-table-cell-align-right oc-table-cell-align-middle oc-table-cell-width-auto oc-text-nowrap oc-table-data-cell oc-table-data-cell-actions">
        <div class="oc-table-files-actions"> <button aria-label="Show details" class="oc-table-files-btn-show-details oc-button oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-raw oc-button-raw-color-default"><span aria-label="icon" class="oc-icon oc-icon-m oc-icon-system" aria-hidden="true">function SvgMore_vert(props) {
  return React.createElement(
    'svg', 
    Object.assign({}, props, {'data-file-name': SvgMore_vert.name})
  );
}</span></button></div>
      </td>
    </tr>
  </tbody>
  <!---->
</table>
`;
